corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js

```diff
@@ -78,9 +83,30 @@ hqDefine("data_dictionary/js/data_dictionary", [
             self.removeFHIRResourcePropertyPath(false);
         };
 
+        self.canHaveAllowedValues = function () {
+            return self.dataType() == 'select';
+        };
+
+        self.allowedValuesDisplay = ko.computed(function() {
```

Comment:
I'd make this a real list, using a `foreach` binding in the HTML and a `list-unstyled` class to avoid bullets, rather than br-separated text, both to be a tiny bit more semantic and to avoid needing to escape the values, which are an HTML injection risk.